tests/unit/test_prestoclient.py

```diff
@@ -94,118 +94,52 @@ def test_http_answer_not_json(self, mock_response,
         type(mock_response.return_value).status =\
             PropertyMock(return_value=200)
         self.assertRaisesRegexp(ValueError, 'No JSON object could be decoded',
-                                client.execute_query, 'any_sql')
-
-    @patch.object(PrestoClient, 'get_response_from')
-    @patch.object(PrestoClient, 'get_next_uri')
-    def test_retrieve_rows(self, mock_uri, mock_get_from_uri, mock_presto_config):
```

Comment:
Generally I would not much believe in mock testing of PrestoClient. I think the best would be to test `PrestoClient` against the real Presto server. There are many quirks that are very difficult to test by mock, which I think would be much easier if we could use real server. Presto Server API may change and here we would not notice this until product tests anyway.